client/v2/internal/util/util.go

```diff
@@ -38,19 +35,21 @@ func ResolveMessageType(resolver protoregistry.MessageTypeResolver, descriptor p
 
 // IsSupportedVersion is used to determine in which version of a module / sdk a rpc was introduced.
 // It returns false if the rpc has comment for an higher version than the current one.
-func IsSupportedVersion(input string) bool {
-	return isSupportedVersion(input, buildInfo)
+func IsSupportedVersion(methodDesc protoreflect.MethodDescriptor) bool {
+	return isSupportedVersion(methodDesc, buildInfo)
 }
 
 // isSupportedVersion is used to determine in which version of a module / sdk a rpc was introduced.
 // It returns false if the rpc has comment for an higher version than the current one.
 // It takes a buildInfo as argument to be able to test it.
-func isSupportedVersion(input string, buildInfo *debug.BuildInfo) bool {
-	if input == "" || buildInfo == nil {
+func isSupportedVersion(methodDesc protoreflect.MethodDescriptor, buildInfo *debug.BuildInfo) bool {
+	hasVersion := proto.HasExtension(methodDesc.Options(), cosmos_proto.E_MethodAddedIn)
+	if !hasVersion || buildInfo == nil || len(buildInfo.Deps) == 0 {
```

Comment:
Are methods without `AddedIn` opts used in all versions?
If so  we should add some comment here. It took me a while to understand why the method doesn't have a version but is still supported.

Reply:
Yes, when the annotation isn't present we consider it as supported for all version. I'll add a comment 👍